src/main/scala/mockws/FakeWSRequestHolder.scala

```diff
@@ -32,30 +36,33 @@ case class FakeWSRequestHolder(
   implicit val materializer: ActorMaterializer
 ) extends WSRequest {
 
+  override type Self = WSRequest
+  override type Response = WSResponse
+
   private val logger = LoggerFactory.getLogger(getClass)
 
   /* Not implemented. */
-  override val calc: Option[WSSignatureCalculator] = None
-  override val followRedirects: Option[Boolean] = None
-  override val proxyServer: Option[WSProxyServer] = None
-  override val virtualHost: Option[String] = None
+  val calc: Option[WSSignatureCalculator] = None
+  val followRedirects: Option[Boolean] = None
+  val proxyServer: Option[WSProxyServer] = None
+  val virtualHost: Option[String] = None
 
-  def withAuth(username: String, password: String, scheme: WSAuthScheme) =
+  def withAuth(username: String, password: String, scheme: WSAuthScheme) : Self =
     copy(auth = Some((username, password, scheme)))
 
-  def sign(calc: WSSignatureCalculator): WSRequest = this
+  def sign(calc: WSSignatureCalculator): Self = this
 
-  def withFollowRedirects(follow: Boolean) = this
+  def withFollowRedirects(follow: Boolean) : Self = this
```

Comment:
minor, but you mixed `: Self` and ` : Self`.
Please stick to `: Self` for consistency.